### PR TITLE
GitHub Workflow Pipeline

### PR DESCRIPTION
**Pull request by TerrenceMcGuinness-NOAA**
**Description**

Adding a GItHub Workflow for running a global-workflow CI Pipline for checkout-build and running a few test experiments.
A self-hosted "runner" is setup on Orion to run this Pipeline.  All interactions with these Pipelines are administered via GitHub such as viewing results and stop and starting the tests.

This version is a "dispatch" mode meaning owners of the GitHub repo can launch the tests with a button on a given branch.
Adding PR based system has been developed and can also update Labels as does the CRON based BASH system.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Clone and Build tests on WCOSS Dell P3
- [x] Cycled test on Orion
- [x] Forecast-only test on Orion
  
**Checklist**

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Comment by WalterKolczynski-NOAA** [comment]
Do we even need the labels anymore after this? Won't the tests now appear under "checks" with the rest of the GH actions?

**Comment by aerorahul** [comment]
> Do we even need the labels anymore after this? Won't the tests now appear under "checks" with the rest of the GH actions?

There are nuances to this way of doing CI.
It is not supposed to be permanent in this form.

**Review by aerorahul — APPROVED** [review]
looks great.

**Comment by aerorahul** [comment]
Here is the resulting pipeline.
Viewable from Github Action itself.
Screenshot 2023-07-14 at 5 26 01 PM

File: ci/scripts/run-check_ci.sh [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+# Ensure the XML is present for the experiment
+if [[ ! -f "${xml}" ]]; then
+  echo "FATAL ERROR: XML file ${xml} not found in '${pslot}', experiment ${pslot} failed, ABORT!"
+  exit 1
+fi
+
+# Launch experiment
+echo "Launch experiment with Rocoto."
+rocotorun -v "${ROCOTO_VERBOSE:-0}" -w "${xml}" -d "${db}"
+sleep 30
```
Comment (WalterKolczynski-NOAA) [inline ci/scripts/run-check_ci.sh:48]: Are these sleeps necessary after `rocotorun`? I didn't think it returned until it was done. If they are needed, (a) I'm worried 30 may be too short (b) the length should be abstracted to a variable so we can change them all in one place.

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: ci/scripts/run-check_ci.sh (hunk 1 above)
Comment (aerorahul) [reply ci/scripts/run-check_ci.sh:48]: please don't over-engineer.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**